options.go

```diff
@@ -48,7 +47,55 @@ func WithProductionLogger() Option {
 // configurations set meant to be used for development.
 func WithDevelopmentLogger() Option {
 	return func(s *SVC) error {
-		logger, atom := newLogger(s.Name, s.Version, zapcore.DebugLevel)
+		logger, atom := newLogger(
+			zapcore.DebugLevel,
+			zapcore.NewJSONEncoder(zap.NewProductionEncoderConfig()),
+		)
+		logger.With(zap.String("app", s.Name), zap.String("version", s.Version))
```

Comment:
Two things here:
* I think that `With()` doesn't alter the logger but rather return a new modified one
* could we move that into `assignLogger` to make sure all loggers have these fields?